tests/BenchmarkDotNet.IntegrationTests/JitRuntimeValidationTest.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Columns;
+using BenchmarkDotNet.Configs;
+using BenchmarkDotNet.Environments;
+using BenchmarkDotNet.IntegrationTests.Xunit;
+using BenchmarkDotNet.Jobs;
+using BenchmarkDotNet.Running;
+using BenchmarkDotNet.Tests.Loggers;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace BenchmarkDotNet.IntegrationTests
+{
+    public class JitRuntimeValidationTest
+    {
+        protected readonly ITestOutputHelper Output;
+
+        private class PlatformConfig : ManualConfig
+        {
+            public PlatformConfig(Runtime runtime, Jit jit, Platform platform)
+            {
+                Add(new Job(Job.Dry, new EnvMode()
+                {
+                    Runtime = runtime,
+                    Jit = jit,
+                    Platform = platform
+                }));
+            }
+        }
+
+        private const string OkCaption = "// OkCaption";
+        private const string LegacyJitNotAvailableForMono = "// ERROR:  LegacyJIT is requested but it is not available for Mono";
+        private const string RyuJitNotAvailable = "// ERROR:  RyuJIT is requested but it is not available in current environment";
+        private const string ToolchainSupportsOnlyRyuJit = "Currently dotnet cli toolchain supports only RyuJit";
+
+        public JitRuntimeValidationTest(ITestOutputHelper outputHelper)
+        {
+            Output = outputHelper;
+        }
+
+        public static TheoryData<Runtime, Jit, Platform, string> DataForWindows = new TheoryData<Runtime, Jit, Platform, string>
+        {
+            // our CI would need to have Mono installed..
```

Comment:
@Ky7m do you know, is it possible to use the latest Mono version in our appveyor build?

Reply:
Appveyor has outdated version and doesn't have an option to configure a required version. We can add an additional step in appveyor yml config to install mono (via chocolatey), so it's possible. Let me play around and create a separate PR. Does it make sense?